File: python/0027-e-remove-element.py
## https://leetcode.com/problems/remove-element
from typing import List
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        nums_no_val = []
        for n in nums:
            if n != val:
                nums_no_val.append(n)

        for i, n in enumerate(nums_no_val):
            nums[i] = n

        return len(nums_no_val)

if __name__ == '__main__':
    print(Solution().removeElement([0,1,1,2,2,3,4,5,1,2,7,5], 2))
    print(Solution().removeElement([0,1,3,4,5,1,2,7,5], 1))
    print(Solution().removeElement([0,1,1,1,1,1,1,1,1], 1))

File: kotlin/0027-e-remove-element.kt
// https://leetcode.com/problems/remove-element/
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var pointer = -1
        var counterVal = 0

        for (i in 0 until nums.size) {
            if (nums[i] == `val`) {
                if (pointer == -1) {
                    pointer = i
                }
                counterVal++
            } else if (pointer > -1) {
                nums[pointer] = nums[i]
                pointer++
            }
        }

        return nums.size - counterVal
    }
}